### DISCUSSION TITLE
CrashLoopBackOff for all injected pods in AKS

### DISCUSSION THREAD

**Discussion by dfenerski** [linkerd/linkerd2#9358] [Help] [answered]
A basic setup that runs inside minikube fails in AKS. No pods can start, precheck and check commands say all is OK.

The logs from the crashing pods don't seem to share much info too:
<pre>
Defaulted container "linkerd-proxy" out of: linkerd-proxy, insight-engine, linkerd-init (init)
[     0.000664s]  INFO ThreadId(01) linkerd2_proxy::rt: Using single-threaded proxy runtime
[     0.001211s]  INFO ThreadId(01) linkerd2_proxy: Admin interface on 0.0.0.0:4191
[     0.001229s]  INFO ThreadId(01) linkerd2_proxy: Inbound interface on 0.0.0.0:4143
[     0.001232s]  INFO ThreadId(01) linkerd2_proxy: Outbound interface on 127.0.0.1:4140
[     0.001234s]  INFO ThreadId(01) linkerd2_proxy: Tap DISABLED
[     0.001236s]  INFO ThreadId(01) linkerd2_proxy: Local identity is default.default.serviceaccount.identity.linkerd.cluster.local
[     0.001239s]  INFO ThreadId(01) linkerd2_proxy: Identity verified via linkerd-identity-headless.linkerd.svc.cluster.local:8080 (linkerd-identity.linkerd.serviceaccount.identity.linkerd.cluster.local)
[     0.001241s]  INFO ThreadId(01) linkerd2_proxy: Destinations resolved via linkerd-dst-headless.linkerd.svc.cluster.local:8086 (linkerd-destination.linkerd.serviceaccount.identity.linkerd.cluster.local)
[     0.055294s]  INFO ThreadId(02) daemon:identity: linkerd_app: Certified identity: default.default.serviceaccount.identity.linkerd.cluster.local
</pre>

Any help on this is immensely appreciated, I can't find any workaround for this so I have to disable linkerd.

Thanks in advance

Dimitar

**Comment by wmorgan** [comment]
Do the logs from the init container provide anything interesting?

**Reply by dfenerski** [reply]
Thanks for the reply @wmorgan !

Injector and destination pods in linkerd NS do have more interesting logs. Specifically, `os error 111` is mentioned, although I have yet to research what this means. 

Thanks for any info on this in advance

BR
[destination_logs.txt](https://github.com/linkerd/linkerd2/files/9536866/destination_logs.txt)
[injector_logs.txt](https://github.com/linkerd/linkerd2/files/9536867/injector_logs.txt)

**Reply by dfenerski** [reply]
I believe this may be a duplicate of https://github.com/linkerd/linkerd2/issues/8343.

**Comment by dfenerski** [comment, marked as answer]
I am not sure whether or why this is related, but injecting linkerd using `linkerd install --set proxyInit.runAsRoot=true` seem to work for me
